### PR TITLE
Only check out a Redis connection if necessary

### PR DESCRIPTION
**Pull request by stanhu**
In Sidekiq 7, calls to `Sidekiq.redis` checks out a connection from an internal connection pool instead of the connection pool used by the job processors. If a connection is checked out but not used, it's possible that the heartbeat thread won't keep the connection alive before the Redis server client timeout.

To avoid this, update the checks to return earlier if the Redis connection is not needed.

This might help reduce intermittent `Broken pipe` errors as reported in https://github.com/redis-rb/redis-client/issues/119.

### CODE REVIEW

File: lib/sidekiq/cron/job.rb [hunk 1]
```diff
@@ -239,12 +241,11 @@ def self.count
       def self.find name
         # If name is hash try to get name from it.
         name = name[:name] || name['name'] if name.is_a?(Hash)
+        return unless exists? name
 
         output = nil
         Sidekiq.redis do |conn|
-          if exists? name
```
Comment (stanhu) [inline lib/sidekiq/cron/job.rb:245]: I wonder if this could cause a deadlock since `exists?` also checks out a connection.

**Review by markets — APPROVED** [review]
Thanks @stanhu!

File: lib/sidekiq/cron/job.rb (hunk 1 above)
Comment (stanhu) [reply lib/sidekiq/cron/job.rb:245]: Probably not, since `connection_pool` appears to reuse the same connection for a given thread.

**Comment by stanhu** [comment]
@markets Thanks for the review! Could I trouble you to tag a release? Thanks again.

**Comment by stanhu** [comment]
@markets Sorry to bother you again. Do you think you could cut a new release?

**Comment by markets** [comment]
@stanhu I'm preparing a new release 👉🏼 #444

**Pull request merged**